administration/src/bp-modules/self-service/hooks/useCardGeneratorSelfService.tsx

```diff
@@ -35,13 +37,21 @@ type UseCardGeneratorSelfServiceReturn = {
 const useCardGeneratorSelfService = (): UseCardGeneratorSelfServiceReturn => {
   const projectConfig = useContext(ProjectConfigContext)
   const appToaster = useAppToaster()
-  const [selfServiceCard, setSelfServiceCard] = useState(
-    initializeCard(projectConfig.card, undefined, { expirationDate: null })
-  )
+  const [cardQueryParams, setSearchParams] = useSearchParams()
+  const [selfServiceCard, setSelfServiceCard] = useState(() => {
+    const headers = getHeaders(projectConfig)
+    const values = headers.map(header => cardQueryParams.get(header))
+    return initializeCardFromCSV(projectConfig.card, values, headers, undefined, true)
+  })
```

Comment:
:upside_down_face: I really like your implementation. If you feel motivated, feel free to add the same functionality to the `useCardGenerator` hook and remove the old implementation from `AddCardsFrom`

Reply:
i think we should merge this soon to create the next beta artifact and i would prefer to implement this in a separate task